src/flb_input_chunk.c

```diff
@@ -788,16 +804,30 @@ size_t flb_input_chunk_set_limits(struct flb_input_instance *in)
      * After the adjustments, validate if the plugin is overlimit or paused
      * and perform further adjustments.
      */
-    if (flb_input_chunk_is_overlimit(in) == FLB_FALSE &&
-        flb_input_buf_paused(in) && in->config->is_running == FLB_TRUE &&
```

Comment:
why flb_input_buf_paused() was removed ?

Reply:
I changed it for a specific reference to a check to in->mem_buf_status, that way pausing for mem_buf and storage are checked separately. To add back in the flb_input_buf_paused() call I could refactor the entire code to check the pause first then check which paused it either memory or storage.